assets/src/components/detoursTable.tsx

```diff
@@ -1,49 +1,107 @@
+export const DetoursTable = ({ data, status }: DetoursTableProps) => (
+  <Table hover={data !== null} className="c-detours-table">
+    <thead className="u-hide-for-mobile">
+      <tr>
+        <th className="px-3 py-4">Route and direction</th>
+        <th className="px-3 py-4 u-hide-for-mobile">Starting Intersection</th>
+        <th className="px-3 py-4 u-hide-for-mobile">On detour since</th>
+      </tr>
+    </thead>
+    <tbody>
+      <DetourRows data={data} status={status} />
+    </tbody>
+  </Table>
+)
+
+const DetourRows = ({
+  data,
+  status,
+}: {
+  data: SimpleDetour[] | null
+  status: DetoursTableStatus
+}) =>
+  data === null ? (
+    <EmptyDetourRows message={`No ${status} detours.`} />
+  ) : (
+    <PopulatedDetourRows data={data} />
+  )
+
+const PopulatedDetourRows = ({ data }: { data: SimpleDetour[] }) => {
   const epochNowInSeconds = useCurrentTimeSeconds()
 
   return (
-    <Table hover className="c-detours-table">
-      <thead className="u-hide-for-mobile">
-        <tr>
-          <th className="px-3 py-4">Route and direction</th>
-          <th className="px-3 py-4 u-hide-for-mobile">Starting Intersection</th>
-          <th className="px-3 py-4 u-hide-for-mobile">On detour since</th>
-        </tr>
-      </thead>
-      <tbody>
-        {data.map((detour, index) => (
-          <tr key={index}>
-            <td className="align-middle p-3">
-              <div className="d-flex">
-                <RoutePill routeName={detour.route} />
-                <div className="c-detours-table__route-info-text d-inline-block">
-                  <div className="pb-1 fs-4 fw-semibold">{detour.name}</div>
-                  <div className="c-detours-table__route-info-direction fs-6">
-                    {detour.direction}
-                  </div>
+    <>
+      {data.map((detour, index) => (
+        <tr key={index}>
+          <td className="align-middle p-3">
+            <div className="d-flex">
+              <RoutePill routeName={detour.route} />
+              <div className="c-detours-table__route-info-text d-inline-block">
+                <div className="pb-1 fs-4 fw-semibold">{detour.name}</div>
+                <div className="c-detours-table__route-info-direction fs-6">
+                  {detour.direction}
                 </div>
               </div>
-            </td>
-            <td className="align-middle p-3 u-hide-for-mobile">
-              {detour.intersection}
-            </td>
-            <td className="align-middle p-3 u-hide-for-mobile">
-              {timeAgoLabel(epochNowInSeconds, detour.updatedAt)}
-            </td>
-          </tr>
-        ))}
-      </tbody>
-    </Table>
+            </div>
+          </td>
+          <td className="align-middle p-3 u-hide-for-mobile">
+            {detour.intersection}
+          </td>
+          <td className="align-middle p-3 u-hide-for-mobile">
+            {timeAgoLabel(epochNowInSeconds, detour.updatedAt)}
```

Comment:
issue(non-blocking): technically this is incorrect semantics and I think would readback strangely to a screen reader, but if you're okay with that we can keep it till it's an issue.

Reply:
I think Github is rendering this weird as well, but I'm reasonably certain that the part of the PR that it's currently highlighting is a pure refactor, and shouldn't change the DOM at all.

Reply:
https://github.com/mbta/skate/pull/2777/files#r1756748196